ui/src/app/store/general/types/base.ts

```diff
@@ -121,6 +121,21 @@ export type KnownArchitecturesState = {
   loading: boolean;
 };
 
+export type KnownBootArchitecture = {
+  arch_octet: string | null;
+  bios_boot_method: string;
+  bootloader_arches: string;
+  name: string;
+  protocol: string;
```

Comment:
I think this could be an enum with "http" and "tftp". https://github.com/maas/maas/blob/master/src/maasserver/websockets/handlers/general.py#L234

Reply:
Good suggestion!